packages/medusa/src/loaders/helpers/routing/index.ts

```diff
@@ -508,35 +727,14 @@ export class RoutesLoader {
           message: `Registering middleware [${route.method}] - ${route.matcher}`,
         })
 
-        this.app[route.method!.toLowerCase()](
+        this.router[route.method!.toLowerCase()](
```

Comment:
suggestion: line 717 I would suggest to do
```ts
const methods = Array.isArray(route.method) ? route.method : [route.method]
for ... of
```

that way you can simplify the code by always managing a array and remove the if...else